core/chains/evm/txm/txm.go

```diff
@@ -130,52 +110,51 @@ func NewTxm(lggr logger.Logger, chainID *big.Int, client Client, attemptBuilder
 
 func (t *Txm) Start(ctx context.Context) error {
 	return t.StartOnce("Txm", func() error {
+		tm, err := NewTxmMetrics(t.chainID)
+		if err != nil {
+			return err
+		}
+		t.metrics = tm
 		t.stopCh = make(chan struct{})
 
 		addresses, err := t.keystore.EnabledAddressesForChain(ctx, t.chainID)
 		if err != nil {
 			return err
 		}
 		for _, address := range addresses {
-			err := t.startAddress(ctx, address)
-			if err != nil {
-				return err
-			}
+			t.startAddress(address)
 		}
 		return nil
 	})
 }
 
-func (t *Txm) startAddress(ctx context.Context, address common.Address) error {
+func (t *Txm) startAddress(address common.Address) {
 	triggerCh := make(chan struct{}, 1)
 	t.triggerCh[address] = triggerCh
-	pendingNonce, err := t.pollForPendingNonce(ctx, address)
-	if err != nil {
-		return err
-	}
-	t.setNonce(address, pendingNonce)
 
 	t.wg.Add(2)
 	go t.broadcastLoop(address, triggerCh)
 	go t.backfillLoop(address)
-	return nil
 }
 
-func (t *Txm) pollForPendingNonce(ctx context.Context, address common.Address) (pendingNonce uint64, err error) {
+func (t *Txm) initializeNonce(ctx context.Context, address common.Address) {
 	ctxWithTimeout, cancel := context.WithTimeout(ctx, pendingNonceDefaultTimeout)
 	defer cancel()
 	for {
-		pendingNonce, err = t.client.PendingNonceAt(ctxWithTimeout, address)
+		pendingNonce, err := t.client.PendingNonceAt(ctxWithTimeout, address)
```

Comment:
Since this function is called repeatedly within broadcastLoop(), it would be efficient if we check here first if nonce is already set for this address. If yes, then return early.

Reply:
This is intentional. Nonce won't have been stored unless the TXM has restarted, in which case we actually do want to overwrite the nonce.